src/ProcessTools.cpp

```diff
@@ -57,44 +77,369 @@ SUPPRESS_VC_WARNING(4548) // This was added to suppress a warning in MSVC's impl
 #include <fstream>
 #include <limits>
 
-RESTORE_WARNING_STATE
+namespace {
+    using namespace Mezzanine;
+#ifdef MEZZ_Windows
+    /// @brief A struct with basic info that can be returned from attempting to launch a process.
+    struct MEZZ_LIB ProcessInfo
+    {
+        /// @brief A HANDLE belonging to the output end of the child processes cout pipe.
+        HANDLE ChildPipe = 0;
+        /// @brief A HANDLE belonging to the child process itself.
+        HANDLE ChildProcess = 0;
+        /// @brief The error code given if the process failed to launch.
+        DWORD ErrorNum = 0;
+        /// @brief The error string given if the process failed to launch.
+        String ErrorStr;
+    };//ProcessInfo
 
-namespace Mezzanine
-{
-    namespace Testing
+    /// @brief Convenience type because Windows likes fatty Strings.
+    using WideString = std::wstring;
+
+    /// @brief Demotes a size_t to an int, which is required by the Win API.
+    /// @brief ToDemote The large integral type to be demoted.
+    /// @return Returns an int of the same value if it can be stored, or asserts otherwise.
+    [[nodiscard]]
+    int IntDemote(const size_t ToDemote)
+    {
+        assert( ToDemote <= static_cast<size_t>( std::numeric_limits<int>::max() ) );
+        return static_cast<int>(ToDemote);
+    }
```

Comment:
This feels like it belongs in a more accessible place, but there are a ton of technical reasons why this is here. And we don't want game devs to need to deal with specific shit for one platform.

Reply:
You mentioned this before when you reviewed it in a different code base.  I can see it being included in Foundation, which would still require a separate implementation here.